### PR TITLE
fix(sample column dependence for when parsing sample internal ids from sample sheets)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR implements a common way to extract sample internal ids from sample sheets accross different CG modules.

This removes dependence on the Sample_Project column in the sample sheet.

**read_sample_sheet.py:**
- Line 41: The regular expression pattern in the `is_valid_sample_internal_id` function was modified. It now includes the start of the string (`^`) to ensure that the pattern matches from the beginning.
- Line 111: The parameter name `flow_cell_type` was changed to `flow_cell_sample_type` for better clarity.

**bcl_convert.py:**
- Line 43: The `sample_project` field in the `BclConvertDemuxMetrics` class was removed.
- Line 62: The `sample_project` field in the `BclConvertAdapterMetrics` class was removed.
- Line 77: The `sample_project` field in the `BclConvertSampleSheetData` class was removed.

**bcl_convert.py:**
- Line 7: The import statement for `SampleSheetNovaSeq6000Sections` from `cg.constants.demultiplexing` was removed.
- Line 12: The import statement for `SAMPLE_SHEET_FILE_NAME` from `cg.constants.bcl_convert_metrics` was removed.
- Line 30: The import statement for `INDEX_CHECK` and `UNDETERMINED` from `cg.constants.demultiplexing` was removed.
- Line 33: The import statement for `BclConvertSampleSheetData` from `cg.apps.sequencing_metrics_parser.models.bcl_convert` was removed.
- Line 35: The import statement for `is_valid_sample_internal_id` from `cg.apps.demultiplex.sample_sheet.read_sample_sheet` was added.

1. In the `__init__.py` file:
   - Removed the `sample_sheet_path` attribute and its assignment.
   - Removed the `parse_sample_sheet_file` method and the calls to it.
   - Removed the check for the existence of the metrics file in the `parse_metrics_file` method.
   - Replaced the call to `is_valid_sample` method with `is_valid_sample_internal_id` in the `get_sample_internal_ids` method.
   - Replaced the call to `get_flow_cell_name` method with direct access to the `flow_cell_name` attribute in the `calculate_total_reads_for_sample_in_lane` method.

2. In the `bcl_convert_to_sequencing_statistics.py` file:
   - Added the `get_flow_cell_name_from_directory` function, which extracts the flow cell name from a directory name.
   - Replaced the call to `get_flow_cell_name` method with `get_flow_cell_name_from_directory` in the `create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell` function.

3. In the `demux_post_processing.py` file:
   - Replaced the call to `get_sample_internal_ids_from_flow_cell` function with `get_sample_internal_ids_from_sample_sheet` in the `update_sample_read_counts_in_status_db` and `add_sample_fastq_files_to_housekeeper` functions, passing `sample_sheet_path` and `flow_cell_sample_type` as arguments.
   - Replaced the call to `get_sample_sheet_path` with `get_sample_sheet_path(flow_cell_directory=flow_cell_directory)` in the `add_sample_sheet_path_to_housekeeper` function.

4. In the `utils.py` file:
   - Removed the import statement and usage of `is_valid_sample_id`.

Certainly! Here's a summary of the changes made to each file:

1. In the `__init__.py` file:
   - Removed the `get_sample_sheet_path` function and replaced it with a new implementation that uses `get_file_in_directory`.
   - Removed the import statement and usage of `is_valid_sample_id`.

2. In the `validation.py` file:
   - Removed the `is_valid_sample_id` function.

5. In the `files.py` file:
   - Added a `FileNotFoundError` exception to be raised when a file is not found in a directory.

6. In the `test_read_sample_sheet.py` file:
   - Updated the `get_sample_internal_ids_from_sample_sheet` function call by replacing the `flow_cell_type` argument with `flow_cell_sample_type`.

7. In the `test_bclconvert.py` file:
   - Removed the `test_parse_bcl_convert_sample_sheet` and `test_is_valid_sample_project` functions.
   - Updated the `test_parse_bcl_convert_metrics` function to remove the assertions related to the `sample_sheet` attribute.
   - Updated the `test_get_flow_cell_name` function to remove the assertion related to the `get_flow_cell_name` method.
   - Removed the `test_is_valid_sample_project` function.

Here are the summarized changes per file:

**demultiplexing.py**:
- The function `get_sample_internal_ids_from_flow_cell` has been removed.
- The function `get_sample_sheet_path` has been modified to accept an optional `sample_sheet_file_name` parameter.

**validation.py**:
- The function `is_valid_sample_id` has been removed.

**files.py**:
- The function `get_file_in_directory` has been modified to raise a `FileNotFoundError` if the file is not found.

**test_read_sample_sheet.py**:
- The parameter name `flow_cell_type` has been changed to `flow_cell_sample_type`.

**test_bclconvert.py**:
- The import statement `import cg.apps.demultiplex.sample_sheet.models` has been added.
- The assertion for `bcl_convert_metrics_parser.sample_sheet` has been removed.

**conftest.py**:
- The fixture name `bcl_convert_demultiplexed_flow_cell_sample_ids` has been changed to `fixture_bcl_convert_demultiplexed_flow_cell_sample_internal_ids`.
- The fixture name `bcl2fastq_demultiplexed_flow_cell_sample_ids` has been changed to `fixture_bcl2fastq_demultiplexed_flow_cell_sample_internal_ids`.
- The function name `store_with_demultiplexed_samples` has been added.
- The fixture `bcl_convert_demultiplexed_flow_cell_sample_ids` and `bcl2fastq_demultiplexed_flow_cell_sample_ids` have been modified to `bcl_convert_demultiplexed_flow_cell_sample_internal_ids` and `bcl2fastq_demultiplexed_flow_cell_sample_internal_ids`, respectively.

**test_demux_post_processing.py**:
- The import `patch` statement has been added.
- The function `test_add_fastq_files_without_sample_id` has been modified to remove the mocked function `get_sample_id_from_sample_fastq_path`.

**test_utils.py**:
- The function `test_get_sample_ids_from_sample_sheet` has been removed.

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/demultiplex/utils.py [hunk 1]
```diff
@@ -74,39 +74,19 @@ def create_delivery_file_in_flow_cell_directory(flow_cell_directory: Path) -> No
...
-            LOG.info(f"Found sample sheet in {directory_path}")
-            return Path(directory_path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
-
-    raise FileNotFoundError(
-        f"Sample sheet not found in given flow cell directory: {flow_cell_directory}"
+def get_sample_sheet_path(
+    flow_cell_directory: Path,
+    sample_sheet_file_name: str = DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+) -> Path:
+    """Get the path to the sample sheet in the flow cell directory."""
```
Comment (diitaz93) [inline cg/meta/demultiplex/utils.py:85]:
```suggestion
    """Return the path to the sample sheet in the flow cell directory."""
```

**Review by diitaz93 — APPROVED** [review]
👍 Nice cleanup

**Comment by ChrOertlin** [comment]
NovaseqX flow cell tests passed altered code (i.e. bcl metrics are being parsed)
cg demultiplex finish temporary 20230508_LH00188_0003_A22522YLT3
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Instantiating cgstats api
Set environment to stage
Finish flow cell 20230508_LH00188_0003_A22522YLT3
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Flow cell already exists in status db: 22522YLT3. Skipping.
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/20230508_LH00188_0003_A22522YLT3/Reports/Quality_Metrics.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/20230508_LH00188_0003_A22522YLT3/Reports/Demultiplex_Stats.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/20230508_LH00188_0003_A22522YLT3/Reports/Adapter_Metrics.csv
Added sequencing metrics to status db for: 22522YLT3
Failed to store flow cell data: <Sequencers.NOVASEQX: 'novaseqx'>
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_main/bin/cg", line 8, in <module>
    sys.exit(base())
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/cli/demultiplex/finish.py", line 65, in finish_flow_cell_temporary
    flow_cell_directory_name=flow_cell_directory_name
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/demultiplex/demux_post_processing.py", line 113, in finish_flow_cell_temp
    self.store_flow_cell_data(parsed_flow_cell)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/demultiplex/demux_post_processing.py", line 124, in store_flow_cell_data
    self.update_sample_read_counts_in_status_db(parsed_flow_cell)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/demultiplex/demux_post_processing.py", line 169, in update_sample_read_counts_in_status_db
    q30_threshold: int = get_q30_threshold(flow_cell_data.sequencer_type)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/demultiplex/utils.py", line 78, in get_q30_threshold
    return FLOWCELL_Q30_THRESHOLD[sequencer_type]
KeyError: <Sequencers.NOVASEQX: 'novaseqx'>

**Comment by ChrOertlin** [comment]
# Novaseq6000 passes tests - confirms working as intended.

cg demultiplex finish temporary 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Instantiating cgstats api
Set environment to stage
Finish flow cell 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Flow cell added to status db: H5F5YDSX7.
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Quality_Metrics.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Demultiplex_Stats.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Adapter_Metrics.csv
Added sequencing metrics to status db for: H5F5YDSX7
Validate that samples are unique in lane 1
Validate that samples are unique in lane 2
Validate that samples are unique in lane 3
Validate that samples are unique in lane 4
Add flow cell data to Housekeeper for H5F5YDSX7
Fetching bundle with name: H5F5YDSX7
Fetching tag with name: fastq
Fetching tag with name: samplesheet
Fetching tag with name: H5F5YDSX7
Validate that samples are unique in lane 1
Validate that samples are unique in lane 2
Validate that samples are unique in lane 3
Validate that samples are unique in lane 4
Fetching bundle with name: ACC12198A10
Fetching bundle with name: ACC12190A3
Fetching bundle with name: ACC11946A30
Fetching bundle with name: ACC12066A94
Fetching bundle with name: ACC12190A8
Fetching bundle with name: ACC12198A9
Fetching bundle with name: ACC12178A6
Fetching bundle with name: ACC12190A2
Fetching bundle with name: ACC12200A6
Fetching bundle with name: ACC12167A2
Fetching bundle with name: ACC12200A4
Fetching bundle with name: ACC11946A22
Fetching bundle with name: ACC12215A3
Fetching bundle with name: ACC12178A7
Fetching bundle with name: ACC12215A1
Fetching bundle with name: ACC12167A1
Fetching bundle with name: ACC12190A5
Fetching bundle with name: ACC12215A5
Fetching bundle with name: ACC12215A4
Fetching bundle with name: ACC12190A6
Fetching bundle with name: ACC12200A5
Fetching bundle with name: ACC12198A8
Fetching bundle with name: ACC12066A51
Fetching bundle with name: ACC11919A7
Fetching bundle with name: ACC12114A1
Fetching bundle with name: ACC12193A4
Fetching bundle with name: ACC12178A4
Fetching bundle with name: ACC11946A18
Fetching bundle with name: ACC12190A7
Fetching bundle with name: ACC12190A4
Fetching bundle with name: ACC12066A34
Fetching bundle with name: ACC12215A2
Fetching bundle with name: ACC12190A1
Fetching bundle with name: ACC12205A1
Fetching bundle with name: ACC12066A39
Fetching bundle with name: ACC12178A8
Fetching bundle with name: ACC11920A11
Fetching bundle with name: ACC12198A11
Fetching bundle with name: ACC11946A34

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def validate_samples_unique_per_lane(samples: List[FlowCellSample]) -> None:
 
 def is_valid_sample_internal_id(sample_internal_id: str) -> bool:
     """Check if a sample internal id has the correct structure."""
```
Comment (ChrOertlin) [inline cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:41]:
```suggestion
    """Check if a sample internal id has the correct structure, startswith three letters followed by at least three digits."""
```

File: cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py [hunk 3]
```diff
@@ -37,3 +37,8 @@ def create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell
             )
 
     return sample_lane_sequencing_metrics
+
+
+def get_flow_cell_name_from_directory(flow_cell_dir: Path) -> str:
+    """Get the flow cell name from the directory name."""
+    return flow_cell_dir.name.split("_")[-1][1:]
```
Comment (diitaz93) [inline cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py:44]: I think this should be a property of the flow cell directory. It already has an implemented property called `split_flow_cell_name`, you could also use that to compile with DRY. In summary I would suggest either using:
1. flow_cell_dir.split_flow_cell_name[0]
2. create a new property in FlowCellDirectory class that outputs this

**Review by diitaz93 — COMMENTED** [review]
I am missing the review of the tests, you can start addressing my other comments

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py [hunk 4]
```diff
@@ -38,8 +38,8 @@ def validate_samples_unique_per_lane(samples: List[FlowCellSample]) -> None:
 
 
 def is_valid_sample_internal_id(sample_internal_id: str) -> bool:
-    """Check if a sample internal id has the correct structure."""
-    return bool(re.search(r"[A-Za-z]{3}\d{3}", sample_internal_id))
+    """Check if a sample internal id has the correct structure, startswith three letters followed by at least three digits."""
```
Comment (Vince-janv) [inline cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:41]:
```suggestion
    """Check if a sample internal id has the correct structure, starts with three letters followed by at least three digits."""
```

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py [hunk 5]
```diff
@@ -110,14 +110,15 @@ def get_samples_by_lane(
 
 
 def get_sample_internal_ids_from_sample_sheet(
-    sample_sheet_path: Path, flow_cell_type: Type[FlowCellSample]
+    sample_sheet_path: Path, flow_cell_sample_type: Type[FlowCellSample]
```
Comment (Vince-janv) [inline cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:113]: What does `Type` do?

File: tests/conftest.py [hunk 6]
```diff
@@ -990,17 +990,17 @@ def fixture_sample_sheet_context(cg_context: CGConfig, lims_api: LimsAPI) -> CGC
     return cg_context
 
 
-@pytest.fixture
-def bcl_convert_demultiplexed_flow_cell_sample_ids() -> List[str]:
+@pytest.fixture(name="bcl_convert_demultiplexed_flow_cell_sample_internal_ids")
```
Comment (diitaz93) [inline tests/conftest.py:993]: Is there a reason why you didn't add the scope fo this fixture?

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py (hunk 5 above)
Comment (Vince-janv) [reply cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:113]: Do we need to pass `flow_cell_sample_type`? Is it not more logical to be able to just give a sample_sheet and have the function figure out what type it is?

File: cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py [hunk 7]
```diff
@@ -77,47 +70,16 @@ def parse_metrics_file(
...
-            for line in sample_sheet_content:
-                sample_sheet_sample_lines.append(BclConvertSampleSheetData(**line))
-        return sample_sheet_sample_lines
-
     def get_sample_internal_ids(self) -> List[str]:
         """Return a list of sample internal ids."""
         sample_internal_ids: List[str] = []
         for sample_demux_metric in self.demux_metrics:
-            if self.is_valid_sample(sample_internal_id=sample_demux_metric.sample_internal_id):
+            if is_valid_sample_internal_id(
```
Comment (Vince-janv) [inline cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py:77]: Nice 🔥

File: tests/meta/demultiplex/tests/fixtures/apps/demultiplexing/flow-cell-runs/nova_seq_6000/201203_A00689_0200_AHVKJCDRXX/copycomplete.txt
Comment (diitaz93) [inline tests/meta/demultiplex/tests/fixtures/apps/demultiplexing/flow-cell-runs/nova_seq_6000/201203_A00689_0200_AHVKJCDRXX/copycomplete.txt:1]: According to the class `DemultiplexingDirsAndFiles` in `cg/constants/demultiplexing.py` this file should be named `CopyComplete.txt` with capital letters as it is a NovaSeq6000 flow cell. Better double check

**Review by diitaz93 — COMMENTED** [review]
Looks good!

File: cg/meta/demultiplex/utils.py [hunk 8]
```diff
@@ -74,39 +74,19 @@ def create_delivery_file_in_flow_cell_directory(flow_cell_directory: Path) -> No
...
+    sample_sheet_path: Path = get_file_in_directory(
+        directory=flow_cell_directory, file_name=sample_sheet_file_name
     )
+    return sample_sheet_path
```
Comment (Vince-janv) [inline cg/meta/demultiplex/utils.py:89]: Can be returned immediately
```suggestion
    return get_file_in_directory(
        directory=flow_cell_directory, file_name=sample_sheet_file_name
    )
```

File: cg/store/api/find_business_data.py [hunk 9]
```diff
@@ -330,6 +330,16 @@ def get_number_of_reads_for_sample_passing_q30_threshold(
         reads_count: Optional[int] = total_reads_query.scalar()
         return reads_count if reads_count else 0
 
+    def get_sample_lane_sequencing_metrics_by_flow_cell_name(
```
Comment (Vince-janv) [inline cg/store/api/find_business_data.py:333]: This looks a bit off topic for the PR since it is not used. Can you extract it to another PR? (same for its tests and filters)

**Review by Vince-janv — APPROVED** [review]
Nice to remove more code than you added ⭐ Have a look and see if `get_sample_internal_ids_from_sample_sheet` can be reworked a bit

File: cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py (hunk 3 above)
Comment (ChrOertlin) [reply cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py]: I removed the function and added a flow cell name attribute to the flow cell directory data class. Furthermore, I renamed some attributes and functions that make it clearer that a flow cell directory name is being parsed from which we then extract the flow cell name

File: tests/conftest.py (hunk 6 above)
Comment (ChrOertlin) [reply tests/conftest.py:993]: I just refactored the name of the fixture. No scope was added prior. Should I add it?

File: tests/meta/demultiplex/tests/fixtures/apps/demultiplexing/flow-cell-runs/nova_seq_6000/201203_A00689_0200_AHVKJCDRXX/copycomplete.txt
Comment (ChrOertlin) [reply tests/meta/demultiplex/tests/fixtures/apps/demultiplexing/flow-cell-runs/nova_seq_6000/201203_A00689_0200_AHVKJCDRXX/copycomplete.txt:1]: I think that file got created somehow while running tests... I did not add it on purpose. I will remove it.

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py (hunk 5 above)
Comment (ChrOertlin) [reply cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:113]: The sample sheet only hold information on the samples and not the sequencer or sequencer type. I think from that alone it will be difficult to know what flow cell sample type is used. I tried to conform to how samples are parsed upstream when reading sample sheets in this module, that is why the usage of the flow cell sample type is there.

Do you think it an issue to have it like this?

File: cg/store/api/find_business_data.py (hunk 9 above)
Comment (ChrOertlin) [reply cg/store/api/find_business_data.py:333]: I added these functions since in a test sql alchemy syntax was used outside of filter functions. We spent a lot of time prior in this year to not have that anymore. I can extract it if needed, on the other hand it is implemented and tested in this pr now.

Are you sure an additional PR is needed for this?

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py (hunk 5 above)
Comment (Vince-janv) [reply cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:113]: So the sample sheets differ between types? If so the information is implicitly there right?

File: cg/store/api/find_business_data.py (hunk 9 above)
Comment (ChrOertlin) [reply cg/store/api/find_business_data.py:333]: https://github.com/Clinical-Genomics/cg/pull/2253

File: cg/apps/demultiplex/sample_sheet/read_sample_sheet.py (hunk 5 above)
Comment (ChrOertlin) [reply cg/apps/demultiplex/sample_sheet/read_sample_sheet.py:113]: I think the other way would be to have a check for column naming in the sample sheet which would be similar to what to the logic from the flow cell sample type. I will open a new PR that opens this possibility and then we can decide there.

**Comment by ChrOertlin** [comment]
Stage:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj
done.
Log deploy... done.
cg, version 38.2.8

Prod:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj
done.
Log deploy... done.
cg, version 38.2.8
